src/Chapter3.hs

```diff
@@ -971,6 +1083,27 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday
+  deriving (Bounded, Enum, Eq, Show)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sunday = Monday
+nextDay d = succ d -- I can't eta reduce here?!
```

Comment:
Unfortunately, no :disappointed: This is Haskell limitation. If you pattern match on N arguments in the first function case, then each case should have the same number of arguments

Reply:
That's a pity. I didn't know that before